config/v1/types_network.go

```diff
@@ -6,27 +6,28 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// Network holds cluster-wide information about Network.  The canonical name is `cluster`
-// TODO this object is an example of a possible grouping and is subject to change or removal
+// Network holds cluster-wide information about Network. The canonical name is `cluster`. It is used to configure the desired network configuration, such as: IP address pools for services/pod IPs, network plugin, etc. 
+// Please view network.spec for an explanation on what is applies when configuring this resource. 
 type Network struct {
 	metav1.TypeMeta `json:",inline"`
 	// Standard object's metadata.
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// spec holds user settable values for configuration.
+	// NetworkSpec is the desired network configuration.
+	// As a general rule, this SHOULD NOT be read directly. Instead, you should
+	// consume the NetworkStatus, as it indicates the currently deployed configuration.
+	// Currently, changing ClusterNetwork, ServiceNetwork, or NetworkType after
+	// installation is not supported.
```

Comment:
Drop the "`NetworkSpec is the desired network configuration.`" line

Having thought about this some more, I don't think we should be listing the immutable fields here. People will forget to change the comment here when they add new fields in the struct below, etc. Instead, maybe say something about "most" of the fields being immutable after installation here, and clarify the specifics on the individual fields in `NetworkSpec`.